### ISSUE TITLE
KAS-FFC noKdfNoKc Issue

### ISSUE THREAD

**Issue by jbarnesweb** [usnistgov/ACVP-Server#107]
https://pages.nist.gov/ACVP/draft-fussell-acvp-kas-ffc.html#name-example-kas-ffc-component-c

Using this capabilities object:

```
			{
				"algorithm": "KAS-FFC",
				"mode": "Component",
				"revision": "1.0",
				"prereqVals": [
					{
						"algorithm": "DSA",
						"valValue": "12345"
					},
					{
						"algorithm": "DRBG",
						"valValue": "self"
					},
					{
						"algorithm": "SHA",
						"valValue": "self"
					}
				],
				"function": [ "fullVal" ],
				"scheme": {
					"dhStatic": {
						"kasRole": [ "initiator" ],
						"noKdfNoKc": {
							"parameterSet": {
								"fc": {
									"hashAlg": [ "SHA2-256" ]
								}
							}
						}
					}
				}
			},

```

I received these test vectors. (Narrowed to one case to illustrate failure)

```
    "algorithm": "KAS-FFC",
    "isSample": false,
    "mode": "Component",
    "revision": "1.0",
    "testGroups": [
      {
        "g": "57C6E2DC6968842F8ECEC84AB0BAF82F63C9973FA7EF818BE61778359569FCC6E1C565F56A77154A319C1058F0C4C94DD32D4BED544DB4A05C45B59171FEFBAC5E2160FDDB8AAD6322C62DD40EAA5125BDB3D67268670E50501BCE5C2FE7DF54D175563D414D173F40DAFBFED14EE68F49DF39BFC1254D1AA85145F6223A0A9EFA0CDA5E1BF3BE2F0C850D0C471D17C835EF11514B5E20DC6B6E87E31639CB27E2452ECDCF620BEFABE0D0BF368B7FCD3EE40832A0ACBEF3B9474D8D50B070FA7F9B1CCAD993FC681A9C440D9406A3F521B5F65758C97ABE6C4B95A99A84E0D86D5A1DCF549C5C024138446D182E464372A420673D6C5F707FE14D198E8C774C",
        "hashAlg": "SHA2-256",
        "kasMode": "noKdfNoKc",
        "kasRole": "initiator",
        "p": "AFE28B16B8BD104E3A287B0C3701443C61D167308395FD9BD1B3F861C789F079F99C7ECD9AF511B5A5B28790104735B01BBBC7E43BD4B5464F28A06A7F7DD03CD648CF932DE0755A79E77A7D22DAE2A3682C76F9F0D8B0B08DBC71446CBA8BD38002D8C17B179B73EE57EB072DA8F913AAEE8271811C538CB300717BC78A82DD10A53B38D6E0A9B804ABEBF80E479A665E5A923533AEE9B2729677B0C4C734536840A58E87B8587C1CA256FFA277E4FABF284DEFB6E45B4C97DF785FA5D66BD9470A783F3DAFA8DDFBDD9B2546DDF8A82AC6BD830C22078853B501AA9B16158D1FC05A036ADDFB3C54A50D00B09D2C847803616A8FA1D57E844B04FFBEB94515",
        "parmSet": "fc",
        "q": "AC4F668A3722A3A1789055336C5097B70F5BB4CA22D874782F086E9CC08971B3",
        "scheme": "dhStatic",
        "testType": "AFT",
        "tests": [
          {
            "staticPublicServer": "383154CC2D9F5208E9F692F27A7047318E0530EC32B5C811C888BC9DA746CF6831F2654B371D62A6034F7B8F6DD12B5EE39895905AC36B7B5423F3259C09C74FF66D1916BB26427DED14F5AA1CC133F12FA8AD48BF8EA40B97815863FBD34A53387A32845B2B64EFED5243DC156B91E029EA475076D8110F0B8A0EF807BAF40947C8C1D11885599A1B8F00498F8B57DB2CB20CFDC3F15C745657AAAA7D201905046F8A67BE63F939685AC0649C4A814ED2A05D6B2859B3E5C879ED2F8563823B0F350668A2D4FD576926D5B6E44202C6AA8AA2DB1FA8F0CB88056D044846D9F93F23A9084DD6E48C045329717F52338234D60451C8CFA57D2733EBC4CC58E271",
            "tcId": 1
          },


```

I provided this result. (Limited to one to illustrate failure).

```

    "vsId": 558790,
    "algorithm": "KAS-FFC",
    "testGroups": [
        {
            "tgId": 1,
            "tests": [
                {
                    "tcId": 1,
                    "staticPublicIut": "8B8C5BE3D485E627FE324B886F2E4C2292C1F5AAE7B5EA570B52C8BFD0A769C4",
                    "hashZIut": "8BAB6F9A069626E04AC11CD1698FFD139D9645FC5364412A3DE65D8CF26EF4C7"
                },

```

I received this failure notice.

```
        "disposition": "failed",
        "tests": [
            {
                "expected": {
                    "tag": "ECF8C2D733CCA1212D6E0071D83A84D1EF2B0FA9DF3EC53C28546E29636F1868"
                },
                "provided": {
                    "hashZ": "8BAB6F9A069626E04AC11CD1698FFD139D9645FC5364412A3DE65D8CF26EF4C7"
                },
                "reason": "HashZ does not match",
                "result": "failed",
                "tcId": 1
            },

```

I'm confused that you expected a tag value for a Component test.

Does the reason mean you expected a HashZ (and it didn't match) or you expected a tag?

Help appreciated.

**Comment by Kritner** [comment]
The `hashZ` is what is expected, `tag` is just an incorrect label being applied to the "expected" object.  Will correct in the next release.

**Comment by jbarnesweb** [comment]
Thank you!

**Comment by Kritner** [comment]
This change is on demo in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18)

**Comment by Kritner** [comment]
This change has been released to prod in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18).
